drivers/adc/adc_ad5592.c

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) 2023 Grinn
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT adi_ad5592_adc
+
+#include <zephyr/drivers/adc.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/drivers/mfd/ad5592.h>
+
+#define ADC_CONTEXT_USES_KERNEL_TIMER
+#include "adc_context.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(adc_ad5592, CONFIG_ADC_LOG_LEVEL);
+
+#define AD5592_ADC_RESOLUTION 12U
+#define AD5592_ADC_MAX_VAL 4096
+
+struct adc_ad5592_config {
+	const struct device *mfd_dev;
+};
+
+struct adc_ad5592_data {
+	struct adc_context ctx;
+	const struct device *dev;
+	uint8_t adc_conf;
+	uint16_t *buffer;
+	uint16_t *repeat_buffer;
+	uint8_t channels;
+	struct k_thread thread;
+	struct k_sem sem;
+
+	K_KERNEL_STACK_MEMBER(stack, CONFIG_ADC_AD5592_ACQUISITION_THREAD_STACK_SIZE);
+};
+
+static int adc_ad5592_channel_setup(const struct device *dev,
+				    const struct adc_channel_cfg *channel_cfg)
+{
+	const struct adc_ad5592_config *config = dev->config;
+	struct adc_ad5592_data *data = dev->data;
+
+	if (k_is_in_isr()) {
```

Comment:
@gmarull I guess this all needs to be removed from the drivers, right?

Reply:
this all depends on the driver class, for example, for some GPIO ops this is required.

Reply:
Okey, removed from ADC and DAC only.